src/Authentication/Authentication.Core/Utilities/AuthenticationHelpers.cs

```diff
@@ -59,16 +63,23 @@ private static async Task<TokenCredential> GetManagedIdentityCredentialAsync(IAu
 
         private static async Task<InteractiveBrowserCredential> GetInteractiveBrowserCredentialAsync(IAuthContext authContext, CancellationToken cancellationToken = default)
         {
+            using AzureEventSourceListener listener = AzureEventSourceListener.CreateConsoleLogger();
             if (authContext is null)
                 throw new AuthenticationException(ErrorConstants.Message.MissingAuthContext);
-
-            var interactiveOptions = new InteractiveBrowserCredentialOptions
+            InteractiveBrowserCredentialOptions interactiveOptions;
+            if (authContext.SigninUi == SigninUi.Native)
             {
-                ClientId = authContext.ClientId,
-                TenantId = authContext.TenantId,
-                AuthorityHost = new Uri(GetAuthorityUrl(authContext)),
-                TokenCachePersistenceOptions = GetTokenCachePersistenceOptions(authContext)
-            };
+                [DllImport("user32.dll")] static extern IntPtr GetForegroundWindow();
+                IntPtr parentWindowHandle = GetForegroundWindow();
+                interactiveOptions = new InteractiveBrowserCredentialBrokerOptions(parentWindowHandle);
+            } else {
+                interactiveOptions = new InteractiveBrowserCredentialOptions();
+            }
+
+            interactiveOptions.ClientId = "307c3ca2-4edb-4b95-b5c4-39601b76b073"; //authContext.ClientId;
+            interactiveOptions.TenantId = "common"; //authContext.TenantId;
```

Comment:
Let's revert this to use the authContext clientId and tenantId.
```suggestion
            interactiveOptions.ClientId = authContext.ClientId;
            interactiveOptions.TenantId = authContext.TenantId;
```

Also, do we need to make changes to the default clientId the SDK ships with to support WAM? If so, please update the PR description with necessary changes needed to support WAM.

Reply:
After reverting back to authContext clientId and tenantId, I got the following error while trying out the WAM integration.

Request Id: 7f5d0508-4131-47b5-bfbc-a0d773240100 
Correlation Id: e1721adf-9c0f-46c6-a798-ac91ad792fb6 
Timestamp: 2022-10-12T18:37:55Z 
Message: AADSTS50011: The redirect URI 'ms-appx-web://Microsoft.AAD.BrokerPlugin/14d82eec-204b-4c2f-b7e8-296a70dab67e' specified in the request does not match the redirect URIs configured for the application '14d82eec-204b-4c2f-b7e8-296a70dab67e'. Make sure the redirect URI sent in the request matches one added to your application in the Azure portal. Navigate to https://aka.ms/redirectUriMismatchError to learn more about how to fix this. 

However, using the current clientId and tenantId in place, I got the following error.
Azure-Identity: False MSAL 4.46.1.0 MSAL.NetCore .NET 6.0.8 Microsoft Windows 10.0.22621 [2022-10-12 18:40:39Z - 7b6b91a8-9534-40d9-9b07-cfccfad4e282] Exception type: System.MissingMethodException

   at Microsoft.Identity.Client.Broker.WamAdapters.ParseRuntimeResponse(AuthResult authResult, AuthenticationRequestParameters authenticationRequestParameters, ILoggerAdapter logger)
   at Microsoft.Identity.Client.Broker.WamAdapters.HandleResponse(AuthResult authResult, AuthenticationRequestParameters authenticationRequestParameters, ILoggerAdapter logger, String errorMessage)
   at Microsoft.Identity.Client.Broker.RuntimeBroker.SignInInteractivelyAsync(AuthenticationRequestParameters authenticationRequestParameters, AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
   at Microsoft.Identity.Client.Broker.RuntimeBroker.AcquireTokenInteractiveAsync(AuthenticationRequestParameters authenticationRequestParameters, AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
   at Microsoft.Identity.Client.Internal.Broker.BrokerInteractiveRequestComponent.FetchTokensAsync(CancellationToken cancellationToken)
   at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.FetchTokensFromBrokerAsync(String brokerInstallUrl, CancellationToken cancellationToken)
   at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.GetTokenResponseAsync(CancellationToken cancellationToken)
   at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.ExecuteAsync(CancellationToken cancellationToken)
   at Microsoft.Identity.Client.Internal.Requests.RequestBase.RunAsync(CancellationToken cancellationToken)
[Informational] Azure-Identity: InteractiveBrowserCredential.Authenticate was unable to retrieve an access token. Scopes: [ User.Read ] ParentRequestId:  Exception: Azure.Identity.AuthenticationFailedException (0x80131500): InteractiveBrowserCredential authentication failed: Method not found: 'System.String Microsoft.Identity.Client.NativeInterop.Account.get_Id()'.

I think it is safe to assume some work still needs to be done to ensure it is working properly

Reply:
I've resolved the above errors in https://github.com/microsoftgraph/msgraph-sdk-powershell/pull/1344/commits/7bf97fa0d31c42647ec715dc0deb120fed185b4f. However, I'm noticing WAM account picker is not opening after:
- `Connect-MgGraph -SignInUI Native` (Sign in using WAM now works as expected).
- `Disconnect-MgGraph` (Sign out of account)
- `Connect-MgGraph -SignInUI Native` (Sign in again using WAM fails to open WAM account picker).

Reply:
Could that be as a result of the PoC created or is that a bug from the Azure Identity module

Reply:
It is likely a bug with Azure Identity. Please create an issue in their repo with repro steps for them to investigate.